### ISSUE TITLE
Wrong job label on KubeJobFailed alert expression

### ISSUE THREAD

**Issue by pvlltvk** [prometheus-operator/prometheus-operator#1869]
**What did you do?**

Trying to set up alerts for failed jobs with Prometheus Operator

**What did you expect to see?**

I'm having some failed jobs, but expression `kube_job_status_failed{job="kube-state-metrics"}  > 0` doesn't show any result, cause label kube-state-metrics for kube_job_status_failed doesn't exist.

**Environment**

* Prometheus Operator version:

	v0.23.1

* Kubernetes version information:

```
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", GitCommit:"d2835416544f298c919e2ead3be3d0864b52323b", GitTreeState:"clean", BuildDate:"2018-02-09T21:51:54Z", GoVersion:"go1.9.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.6+coreos.0", GitCommit:"c2aac694c2c7373dfe3ad81f47fc3cbc70a5a8fa", GitTreeState:"clean", BuildDate:"2018-03-21T21:54:22Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind: 

	bare-metal

* Manifests:

Same rules manifest as here - [prometheus-rules.yaml](https://github.com/coreos/prometheus-operator/blob/master/contrib/kube-prometheus/manifests/prometheus-rules.yaml)

* Prometheus Operator Logs:

I don't think that Prometheus Operator Logs is needed here :)

**Comment by brancz** [comment]
Because of Prometheus's somewhat reserved use of the `job` label, we [changed the kube-state-metrics label to `job_name`](https://github.com/kubernetes/kube-state-metrics/commit/b8790756c7d064a0797980bc79eba6c28c52bde7#diff-9f524e9f18963b5079856e0848de0811) instead, but this only landed in v1.4.0 and higher. Two options: you can either upgrade kube-state-metrics to v1.4.0 (which we should do regardless), or use the `job_exported` label, which is what Prometheus transforms the original metric's label into because of the `job` label clash.

**Comment by pvlltvk** [comment]
@brancz Ok, thank you for detailed answer
